pkg/execution/planning.go

```diff
@@ -134,14 +134,22 @@ func (p *planningVisitor) LeaveInlineFragment(ref int) {
 }
 
 func (p *planningVisitor) EnterField(ref int) {
+	fieldName := p.operation.FieldNameUnsafeString(ref)
 
 	definition, exists := p.FieldDefinition(ref)
 	if !exists {
-		return
+		if fieldName != "__typename" {
```

Comment:
If you merge base schema and extend base schema to add a `__typename` field to every object & interface type definition (via visitor), you would not have to ugly-fix this I think. What do you think about "cleaning" this up?